huxley/www/js/components/ChairSummaryView.js

```diff
@@ -0,0 +1,185 @@
+/**
+* Copyright (c) 2011-2016 Berkeley Model United Nations. All rights reserved.
+* Use of this source code is governed by a BSD License (see LICENSE).
++*/
+
+'use strict';
+
+var React = require('react');
+var ReactRouter = require('react-router');
+
+var Button = require('components/Button');
+var AssignmentStore = require('stores/AssignmentStore');
+var CountryStore = require('stores/CountryStore');
+var CurrentUserStore = require('stores/CurrentUserStore');
+var DelegateActions = require('actions/DelegateActions');
+var DelegateStore = require('stores/DelegateStore');
+var InnerView = require('components/InnerView');
+var User = require('utils/User');
+
+var ChairSummaryView = React.createClass({
+  mixins: [
+    ReactRouter.History,
+  ],
+
+  getInitialState() {
+    var user = CurrentUserStore.getCurrentUser();
+    return {
+      assignments: AssignmentStore.getCommitteeAssignments(user.committee),
+      countries: CountryStore.getCountries(),
+      country_assignments: {},
+      delegates: DelegateStore.getCommitteeDelegates(user.committee),
+    };
+  },
+
+  componentWillMount() {
+    var user = CurrentUserStore.getCurrentUser();
+    if (!User.isChair(user)) {
+      this.history.pushState(null, '/');
+    }
+  },
+
+  componentDidMount() {
+    var user = CurrentUserStore.getCurrentUser();
+    this._handleMapAssignments();
+    this._delegatesToken = DelegateStore.addListener(() => {
+      this.setState({delegates: DelegateStore.getCommitteeDelegates(user.committee)});
+      this._handleMapAssignments();
+    });
+
+    this._assignmentsToken = AssignmentStore.addListener(() => {
+      this.setState({assignments: AssignmentStore.getCommitteeAssignments(user.committee)});
+      this._handleMapAssignments();
+    });
+
+    this._countriesToken = CountryStore.addListener(() => {
+      this.setState({countries: CountryStore.getCountries()});
+    });
+  },
+
+  componentWillUnmount() {
+    this._countriesToken && this._countriesToken.remove();
+    this._delegatesToken && this._delegatesToken.remove();
+    this._assignmentsToken && this._assignmentsToken.remove();
+  },
+
+  render() {
+    return (
+      <InnerView>
+        <h2>Summaries</h2>
+        <p>
+          Here you can provide feedback for the delegates. Note that publishing
```

Comment:
We should probably be more explicit here as well: "Please note that clicking the "Publish" button will allow advisors to read the summaries you have written about their delegates."